### DISCUSSION TITLE
HSET Key Expiration

### DISCUSSION THREAD

**Discussion by marksalpeter** [valkey-io/valkey#1028] [Q&A]
Hi Guys! In Redis 7.4 they revamped the key structure to allow TTL in hash sets. Is adding similar functionality a priority for Valkey as well? Its extremely useful functionality for things like read optimized frequency capping.

**Comment by madolson** [comment]
Being discussed here: https://github.com/valkey-io/valkey/issues/640, feel free to add your use case to get it prioritized faster.
